File: Missing_Number.cpp
//Given an array nums containing n distinct numbers in the range [0, n], return the only number in the range that is missing from the array.

class Solution {
public:
    int missingNumber(vector<int>& nums) {
        sort(nums.begin(), nums.end());

        int len = nums.size();
        for (int i = 0; i < len; i++)
        {
            if(nums[i] != i)
                return i;
        } 
        return len;  
    }
};

File: python_track/missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        sum_from_zero_to_n = n * (n + 1) // 2 
        sum_of_nums = sum(nums)
        num = sum_from_zero_to_n - sum_of_nums

        return num
        
        # nums_set = set([i for i in range(len(nums) + 1)])

        # for num in nums:
        #     nums_set.remove(num)
        
        # return list(nums_set)[0]
